src/cryptonote_basic/cryptonote_basic_impl.h

```diff
@@ -74,16 +74,37 @@ struct address_parse_info {
     KV_MAP_SERIALIZABLE
 };
 
+// Strongly-typed money amount used to interface with the SQL DB when inserting
+// money into the DB. Amounts stored in the DB use a higher precision by a
+// factor of `BATCH_REWARD_FACTOR`.
+struct sql_db_money {
```

Comment:
This type safety is a great change!

A tiny bit of bikeshedding though: I think it would be worthwhile renaming this to, perhaps, `reward_money` to better describe what it is used for rather than where it gets stored.  (Yeah, the database stores the * 1000 value, but the real reason for the * 1000 is for reward calculation values well before it hits the db, rather than something for the db itself).

Reply:
Yep renamed